enhancements/etcd/etcd-encyrption-for-separate-oauth-resources.md

```diff
@@ -0,0 +1,102 @@
+It holds an array of keys that are used to encrypt resources. 
+
+OpenShift maintains a single `EncryptionConfiguration` resource that is created and maintained by a set of controllers called the encryption controllers in `openshift-config-managed` namespace for each component that needs encryption.
+
+1. Create a new deployer called `UnionRevisionLabelPodDeployer` that will manage multiple `RevisionLabelPodDeployer`.
+2. Implement `DeployedEncryptionConfigSecret() (secret *corev1.Secret, converged bool, err error)` function that:
+   - calls `DeployedEncryptionConfigSecret` for all components
+   - returns "failure" if any component has reported an error (`err !=nil`) or hasn't yet converged (`converged == false`)
+   - returns "failure" if the `secret` resource differs among components. In oder to compare the `secret`:
+     - use `encryptionconfig.FromSecret` function that returns `EncryptionConfiguration`
+     - use `reflect.DeepEqual` function to compare `EncryptionConfigurration.Resources` for example `reflect.DeepEqual(openshiftAPIServerEncryptionCfg.Resources, oauthAPIServerEncryptionCfg.Resources)`
+   - returns "success" otherwise
+3. Update `OAS-O` to use `UnionRevisionLabelPodDeployer` passing two `RevisionLabelPodDeployer`. The first one for `openshift-apiserver` (already existing) and the second one for `oauth-apiserver`.
+   - the new deployer will sync `encryption-config` secret from `openshift-config-managed` namespace to `openshift-oauth-apiserver` namespace
+   - it will also report whether all instances of `oauth-apiserver` converged to the same revision
+
+### User Stories [optional]
+
+#### Story 1
+
+#### Story 2
+
+### Risks and Mitigations
+
+
+## Design Details
+
+### Test Plan
+
+
+##### Removing a deprecated feature
+
+See upgrade/downgrade.
+
+### Upgrade / Downgrade Strategy
+
+In an upgrade case, `OAS-O` in version `4.n` will be responsible for synchronizing and maintaining encryption state for both `openshift-apiserver` and `oauth-apiserver`.
+Since both will share exactly the same `encryption-config`, the new component will be able to read (decrypt) already encrypted data.
+
+During an upgrade, the new `UnionRevisionLabelPodDeployer` will create back pressure in the system and will make the encryption controllers wait for the new component to synchronize. 
+In that state, the status of `OAS-O` won't change and we won't roll out any new encryption keys.
+
+In a downgrade scenario, `OAS-O` in version `4.(n-1)` is responsible for synchronizing and maintaining encryption state only for `openshift-apiserver`.
+Since in version `4.n` exactly the same `encryption-config` was used the `openshift-apiserver` will be able to read (decrypt) data. 
+
+### Version Skew Strategy
+
+See the upgrade/downgrade strategy.
+
+## Implementation History
+
+Major milestones in the life cycle of a proposal should be tracked in `Implementation
+History`.
+
+## Alternatives
+
+TODO:
```

Comment:
planning to describe that `CAO` could have its own controller that would use exactly the same `encryption-config` but that would require more coordination between the two `OAS-O` and `CAO`